### PR TITLE
Upgrade qiskit to 0.31.0

### LINKED ISSUE #203: [pytket-qiskit] Update to qiskit 0.30.

### PR DESCRIPTION
**Pull request by RolandMacDoland**
Fixes #203.

This PR addresses the upgrade of the `Qiskit` dependency to version 0.31.0 in the `pytket-qiksit` extension. 
All test pass locally.
I also added a `base-test-requirements.txt` that contain common dependencies for testing extensions. I can harmonise across all extensions and update the `README.md` if found useful.

### CODE REVIEW

**Review by ss2165 — DISMISSED** [review]

**Comment by cqc-alec** [comment]
Can you add an entry in the changelog please?

File: modules/pytket-qiskit/tests/test-requirements.txt [hunk 1]
```diff
@@ -1 +1,2 @@
-pytest-timeout ~= 1.4.2 
\ No newline at end of file
+-r ../../../base-test-requirements.txt
```
Comment (cqc-alec) [inline modules/pytket-qiskit/tests/test-requirements.txt:1]: Can we make this a separate PR, changing all extensions at once rather than just this one?
Currently these are installed in the `build-test` script; if we do make this change then we should remove the installation from there.

Comment (cqc-alec) [reply modules/pytket-qiskit/tests/test-requirements.txt:1]: I would prefer this change to be in a separate PR though.
